// @atlaskit/form/dist/cjs/Validator.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = require("react");

/* eslint-disable no-console */
// While in dev preview console.info will be used

/* eslint-disable react/no-unused-prop-types */
var Validator =
/*#__PURE__*/
function (_PureComponent) {
  (0, _inherits2.default)(Validator, _PureComponent);

  function Validator() {
    (0, _classCallCheck2.default)(this, Validator);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Validator).apply(this, arguments));
  }

  (0, _createClass2.default)(Validator, [{
    key: "render",

    /** Valiate a value using the function & options passed in as props  */

    /*validate = (value:string) => {
      let valid:boolean = this.props.func(value, this.props.options);
      // Invert result if validOn prop is false
      //if(!this.props.validOn && !valid) valid = true;
      return {
          valid,
          message: (valid)? this.props.valid : this.props.invalid
      }
    };
    */
    value: function render() {
      return null;
    }
  }]);
  return Validator;
}(_react.PureComponent);

exports.default = Validator;
(0, _defineProperty2.default)(Validator, "defaultProps", {
  func: {},
  validOnFalse: false,
  valid: '',
  invalid: ''
});
