// @atlaskit/form/dist/cjs/ValidatorMessage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("@atlaskit/theme");

var _error = _interopRequireDefault(require("@atlaskit/icon/glyph/error"));

var _success = _interopRequireDefault(require("@atlaskit/icon/glyph/editor/success"));

var IconWrapper = _styledComponents.default.span.withConfig({
  displayName: "ValidatorMessage__IconWrapper",
  componentId: "sc-1jelslt-0"
})(["\n  display: inline-block;\n  vertical-align: text-top;\n"]);

var Error = function Error() {
  return _react.default.createElement(IconWrapper, null, _react.default.createElement(_error.default, {
    size: "small",
    role: "presentation"
  }));
};

var Success = function Success() {
  return _react.default.createElement(IconWrapper, null, _react.default.createElement(_success.default, {
    size: "small",
    role: "presentation"
  }));
};

var Message = _styledComponents.default.div.withConfig({
  displayName: "ValidatorMessage__Message",
  componentId: "sc-1jelslt-1"
})(["\n  ", " font-weight: normal;\n  color: ", ";\n  margin-top: ", "px;\n"], _theme.typography.h200, function (props) {
  return props.invalid ? _theme.colors.R400 : _theme.colors.G400;
}, (0, _theme.gridSize)() / 2);

var ValidatorMessage =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(ValidatorMessage, _Component);

  function ValidatorMessage() {
    (0, _classCallCheck2.default)(this, ValidatorMessage);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(ValidatorMessage).apply(this, arguments));
  }

  (0, _createClass2.default)(ValidatorMessage, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          invalidMessage = _this$props.invalidMessage,
          validMessage = _this$props.validMessage,
          isInvalid = _this$props.isInvalid; // Validation state is invalid, valid or not validated (null)

      if (isInvalid) {
        return _react.default.createElement(Message, {
          invalid: true
        }, _react.default.createElement(Error, null), invalidMessage); // If the field hasn't been validated yet then isInvalid will be undefined
      } else if (this.props.isInvalid !== undefined && validMessage && validMessage.length) {
        return _react.default.createElement(Message, null, _react.default.createElement(Success, null), validMessage);
      }

      return null;
    }
  }]);
  return ValidatorMessage;
}(_react.Component);

exports.default = ValidatorMessage;
(0, _defineProperty2.default)(ValidatorMessage, "defaultProps", {
  invalidMessage: '',
  validMessage: '',
  isInvalid: undefined,
  type: 'field'
});
