// @atlaskit/button/dist/cjs/styled/ButtonContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _react = _interopRequireDefault(require("react"));

var _theme = require("@atlaskit/theme");

var _utils = require("./utils");

/* eslint-disable react/prop-types */

/* eslint-disable react/no-unused-prop-types */
var getAlignment = function getAlignment(p) {
  return p.followsIcon ? 'baseline' : 'center';
};

var gridSizeDiv2 = _theme.math.divide(_theme.gridSize, 2);

var getMargin = function getMargin(p) {
  return p.spacing === 'none' ? 0 : "0 ".concat(gridSizeDiv2(p), "px");
};

var ButtonContent = function ButtonContent(props) {
  var style = (0, _objectSpread2.default)({
    alignItems: getAlignment(props),
    alignSelf: getAlignment(props),
    flex: '1 1 auto',
    margin: getMargin(props),
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
  }, (0, _utils.getLoadingStyle)(props));
  return _react.default.createElement("span", {
    style: style
  }, props.children);
};

var _default = ButtonContent;
exports.default = _default;
