// @atlaskit/button/dist/cjs/styled/LoadingSpinner.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _spinner = _interopRequireDefault(require("@atlaskit/spinner"));

var LoadingDiv = _styledComponents.default.div.withConfig({
  displayName: "LoadingSpinner__LoadingDiv",
  componentId: "xoee7l-0"
})(["\n  display: flex;\n  position: absolute;\n  left: 50%;\n  top: 50%;\n  transform: translate(-50%, -50%);\n"]);

var LoadingSpinner =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(LoadingSpinner, _Component);

  function LoadingSpinner() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, LoadingSpinner);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(LoadingSpinner)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "invertSpinner", function () {
      var _this$props = _this.props,
          appearance = _this$props.appearance,
          isSelected = _this$props.isSelected,
          isDisabled = _this$props.isDisabled;
      if (isSelected) return true;else if (isDisabled) return false;else if (appearance === 'primary' || appearance === 'danger' || appearance === 'help') return true;
      return false;
    });
    return _this;
  }

  (0, _createClass2.default)(LoadingSpinner, [{
    key: "render",
    value: function render() {
      var spacing = this.props.spacing;
      var spinnerSize = 'medium';

      if (spacing !== 'default') {
        spinnerSize = 'small';
      }

      return _react.default.createElement(LoadingDiv, {
        spacing: this.props.spacing
      }, _react.default.createElement(_spinner.default, {
        size: spinnerSize,
        invertColor: this.invertSpinner()
      }));
    }
  }]);
  return LoadingSpinner;
}(_react.Component);

exports.default = LoadingSpinner;
