// @atlaskit/page/dist/cjs/GridColumn.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = require("styled-components");

var _vars = require("./internal/vars");

var _GridColumnElement = _interopRequireDefault(require("./internal/GridColumnElement"));

var _class, _temp;

var defaultSpacing = 'cosy';

var _default = (0, _styledComponents.withTheme)((_temp = _class =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(AkGridColumn, _Component);

  function AkGridColumn() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, AkGridColumn);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(AkGridColumn)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "getTheme", function (props) {
      return {
        columns: props.theme && props.theme.columns ? props.theme.columns : _vars.defaultGridColumns,
        spacing: props.theme && props.theme.spacing ? props.theme.spacing : defaultSpacing,
        isNestedGrid: false
      };
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "getNestedTheme", function (props) {
      return {
        columns: props.medium,
        spacing: props.theme && props.theme.spacing ? props.theme.spacing : defaultSpacing,
        isNestedGrid: true
      };
    });
    return _this;
  }

  (0, _createClass2.default)(AkGridColumn, [{
    key: "render",
    value: function render() {
      return _react.default.createElement(_styledComponents.ThemeProvider, {
        theme: this.getTheme(this.props)
      }, _react.default.createElement(_GridColumnElement.default, {
        medium: this.props.medium
      }, _react.default.createElement(_styledComponents.ThemeProvider, {
        theme: this.getNestedTheme(this.props)
      }, _react.default.createElement("div", null, this.props.children))));
    }
  }]);
  return AkGridColumn;
}(_react.Component), (0, _defineProperty2.default)(_class, "defaultProps", {
  medium: 0
}), _temp));

exports.default = _default;
